### ISSUE TITLE
can't use iteration element for indexing in loops

### ISSUE THREAD

**Issue by mhaupt** [medusa-ui/medusa#155]
If you use a construct like this ...
```
<m:foreach collection="col" eachName="itm">
<m:text item="map[itm]"/>
</m:foreach>
```
... it won't work, because the `itm` variable is not available in the nested evaluation context (`map[itm]`). It is available in the loop's top context (`<m:text item="itm"/>` will work).

Please refer to [this branch in my Medusa fork](https://github.com/mhaupt/medusa/tree/bug/iteration-element-indexing) for some tests to reproduce this.

**Comment by mhaupt** [comment]
It seems the logic for parsing and interpreting expressions is inconsistent for references at the top level and nested levels of expression strings.

**Comment by kevindeyne** [comment]
I believe this is due to logic in ValueTag (specifically getPossibleEachValue in AbstractTag)

https://github.com/medusa-ui/medusa/blob/15d2aa68a3304ea795e299896f2d0bc821cf4c78/medusa-ui/src/main/java/io/getmedusa/medusa/core/injector/tag/AbstractTag.java#L39-L54

It seems like in the case of `variableA[variableB]`, we consider that `variableA` might be the each value. It doesn't look like we ever considered `variableB` might be the each value.

**Comment by kevindeyne** [comment]
I've added a basic fix in [bug/iteration-element-indexing](https://github.com/medusa-ui/medusa/tree/bug/iteration-element-indexing). Not quite sure if there's more to it than that, I think I would have to test out a few more scenarios to be sure.

**Comment by mhaupt** [comment]
Thanks, @kevindeyne - with the latest changes to `websocket.js`, my sample project has just made a huge leap forward!

I'd appreciate if the integration test I contributed could get the proper attribution. Want me to do a PR?

**Comment by kevindeyne** [comment]
@mhaupt Yes, feel free to make a PR. It can be of just the tests you had and I'll figure it out from there.

**Comment by mhaupt** [comment]
Thank you. [Et voilà.](https://github.com/medusa-ui/medusa/pull/159)

**Comment by kevindeyne** [comment]
Thx! Merged and added to 0.3.7 (It takes a while before a new version propagates to all maven repositories; somewhere between 24/48h)

**Comment by mhaupt** [comment]
Thanks!
